app/models/renewal/SendTheLargestAmountsOfMoney.scala

```diff
@@ -17,47 +17,92 @@
 package models.renewal
 
 import jto.validation.forms.UrlFormEncoded
-import jto.validation.{From, Rule, Write}
+import jto.validation.{From, Path, Rule, RuleLike, To, Write, WriteLike}
 import models.Country
-import play.api.libs.json.Json
-
-case class SendTheLargestAmountsOfMoney(
-                       country_1: Country,
-                       country_2: Option[Country] = None,
-                       country_3: Option[Country] = None
-                     ) {
-
-  def countryList = {
-    this.productIterator.collect {
-      case Some(Country(name, code)) => Country(name, code)
-      case x: Country => x
-    }
-  }
+import play.api.libs.json.{Json, Reads, Writes}
+import utils.TraversableValidators
 
-}
+case class SendTheLargestAmountsOfMoney (countries: Seq[Country])
 
-object SendTheLargestAmountsOfMoney {
+private sealed trait SendTheLargestAmountsOfMoney0 {
+
+  private implicit def rule[A]
+  (implicit
+   a: Path => RuleLike[A, Seq[String]],
+   cR: Rule[Seq[String], Seq[Country]]
+  ): Rule[A, SendTheLargestAmountsOfMoney] =
+    From[A] { __ =>
+
+      import TraversableValidators._
+      import utils.MappingUtils.Implicits.RichRule
 
-  implicit val format = Json.format[SendTheLargestAmountsOfMoney]
+      implicit val emptyToNone: String => Option[String] = {
```

Comment:
Where is this used?

Reply:
this implicit is necessary for seqToOptionSeq[String] below